python/src/sdf/pyJoint.cc

```diff
@@ -0,0 +1,110 @@
+         "Set the name of the parent link.")
+    .def("child_link_name", &sdf::Joint::ChildLinkName,
+         "Get the name of this joint's child link.")
+    .def("set_child_link_name", &sdf::Joint::SetChildLinkName,
+         " Set the name of the child link")
+    .def("resolve_child_link", &sdf::Joint::ResolveChildLink,
+         "Resolve the name of the child link from the "
+         "FrameAttachedToGraph.")
+    .def("resolve_parent_link", &sdf::Joint::ResolveParentLink,
+         "Resolve the name of the parent link from the "
+         "FrameAttachedToGraph. It will return the name of a link or "
+         "\"world\".")
+    .def("axis", &sdf::Joint::Axis,
+         pybind11::return_value_policy::reference,
+         "Get a joint axis.")
+    .def("set_axis", &sdf::Joint::SetAxis,
+         "Set a joint axis.")
+    .def("raw_pose", &sdf::Joint::RawPose,
+         "Get the pose of the joint. This is the pose of the joint "
+         "as specified in SDF (<joint> <pose> ... </pose></joint>). "
+         "Transformations have not been applied to the return value.")
+    .def("set_raw_pose", &sdf::Joint::SetRawPose,
+         "Set the pose of the joint.")
+    .def("pose_relative_to", &sdf::Joint::PoseRelativeTo,
+         "Get the name of the coordinate frame relative to which this "
+         "object's pose is expressed. An empty value indicates that the frame "
+         "is relative to the child link frame.")
+    .def("set_pose_relative_to", &sdf::Joint::SetPoseRelativeTo,
+         "Set the name of the coordinate frame relative to which this "
+         "object's pose is expressed. An empty value indicates that the frame "
+         "is relative to the child link frame.")
+    .def("thread_pitch", &sdf::Joint::ThreadPitch,
+         "Get the thread pitch (only valid for screw joints)")
+    .def("set_thread_pitch", &sdf::Joint::SetThreadPitch,
+         pybind11::return_value_policy::reference,
+         "Set the thread pitch (only valid for screw joints)")
+    .def("semantic_pose", &sdf::Joint::SemanticPose,
+         "Get SemanticPose object of this object to aid in resolving "
+         "poses.")
+    .def("__copy__", [](const sdf::Joint &self) {
+      return sdf::Joint(self);
+    })
+    .def("__deepcopy__", [](const sdf::Joint &self, pybind11::dict) {
+      return sdf::Joint(self);
+    }, "memo"_a);
+
+    pybind11::enum_<sdf::JointType>(jointModule, "JointType")
+      .value("INVALID", sdf::JointType::INVALID)
+      .value("BALL", sdf::JointType::BALL)
+      .value("CONTINUOUS", sdf::JointType::CONTINUOUS)
+      .value("FIXED", sdf::JointType::FIXED)
+      .value("GEARBOX", sdf::JointType::GEARBOX)
+      .value("PRISMATIC", sdf::JointType::PRISMATIC)
+      .value("REVOLUTE", sdf::JointType::REVOLUTE)
+      .value("REVOLUTE2", sdf::JointType::REVOLUTE2)
+      .value("SCREW", sdf::JointType::SCREW)
+      .value("UNIVERSAL", sdf::JointType::UNIVERSAL);
```

Comment:
Pending discussion in https://github.com/ignitionrobotics/sdformat/pull/968#discussion_r851576009
```suggestion
      .value("UNIVERSAL", sdf::JointType::UNIVERSAL)
      .export_values();
```